cmd/buildah/common.go

```diff
@@ -158,3 +159,17 @@ func getDockerAuth(creds string) (*types.DockerAuthConfig, error) {
 		Password: password,
 	}, nil
 }
+
+func validateFlags(c *cli.Context, flags []cli.Flag) error {
```

Comment:
I think this definitely needs a comment explaining intent. It took me a good 5 minutes and the comments on this pull request to understand what you were doing here.

Reply:
👍 @mheon comment on comment.

I played around with it a bunch this morning.  This does catch corner cases only from what I could see.  Most of the error checking for the params would pick things off after it would  go through here.  One example that I ran was 
```
buildah inspect --type --type image george
```
The only downside is it goes through this code every single invocation and will probably only catch an error less than 1% of the time.  Without this change in place the above command told me I'd "too many argument specified".  All that said, it is nice when it catches a corner case and the overhead is minimal. 

The one boolean flag that I was thinking about was tls-verify which takes a boolean value.  But if one is not specified, it defaults to True, so I don't think we need to check that.

I'll say LGTM once Matt's comments are addressed.

Reply:
Added comment.